lib/script-loader.php

```diff
@@ -72,6 +91,41 @@ function gutenberg_enqueue_global_styles() {
 add_action( 'wp_enqueue_scripts', 'gutenberg_enqueue_global_styles' );
 add_action( 'wp_footer', 'gutenberg_enqueue_global_styles', 1 );
 
+/**
+ * Adds JS logic to the Customizer preview for live previewing Custom CSS for Block Themes.
+ *
+ * Note: The logic in this function would be back-ported into customize-preview.js.
+ */
+function gutenberg_add_customizer_block_theme_custom_css_preview_js() {
+	if ( ! ( is_customize_preview() && wp_is_block_theme() ) ) {
+		return;
+	}
+
+	$setting_id = 'custom_css[' . get_stylesheet() . ']';
+
+	$js_function = <<<JS
+		( settingId ) => {
+			wp.customize( settingId, function ( setting ) {
+				setting.bind( function ( newValue ) {
+					const style = document.querySelector( 'style#global-styles-inline-css' );
+					if ( ! style ) {
+						return;
+					}
+					newValue = newValue.replace( /\/\*(BEGIN|END)_CUSTOMIZER_CUSTOM_CSS\*\//g, '' ); // Forbid milestone comments from appearing in Custom CSS which would break live preview.
+					style.textContent = style.textContent.replace( /(\/\*BEGIN_CUSTOMIZER_CUSTOM_CSS\*\/)((?:.|\s)*?)(\/\*END_CUSTOMIZER_CUSTOM_CSS\*\/)/, function ( match, beforeComment, oldValue, afterComment ) {
+						return beforeComment + newValue + afterComment;
+					} );
+				} );
+			} )
+		}
+JS;
+	wp_add_inline_script(
+		'customize-preview',
+		sprintf( '( %s )( %s )', $js_function, wp_json_encode( $setting_id ) )
+	);
+}
+add_action( 'wp_enqueue_scripts', 'gutenberg_add_customizer_block_theme_custom_css_preview_js' );
```

Comment:
I haven't tested it to see if it works, but can we use the `customize_preview_init` hook to avoid unnecessary function execution on the frontend? This hook seems to be widely used in default themes to enqueue scripts as well.

Reply:
So like this?

```php
add_action( 'customize_preview_init', static function () {
    add_action( 'wp_enqueue_scripts', 'gutenberg_add_customizer_block_theme_custom_css_preview_js' );
} );
```

This would then eliminate the need for the `is_customize_preview()` check in the function.

Reply:
Actually, there's not really any need to use `wp_enqueue_scripts` here. We're just adding an inline script to a script that already registered. So this can be:

```suggestion
add_action( 'customize_preview_init', 'gutenberg_add_customizer_block_theme_custom_css_preview_js' );
```

Reply:
See d9f7988befa4808af65fed8741eb53dc43b5eb53